### ISSUE TITLE
Remarks contain incorrect information

### ISSUE THREAD

**Issue by hsebs** [MicrosoftDocs/windows-driver-docs-ddi#966]
Following sentence is wrong.
`Thus each call to FltFastIoPrepareMdlWrite must be followed by a call to FltFastIoMdlWriteComplete.`

FltFastIoMdlWriteComplete must be called when FltFastIoPrepareMdlWrite allocated mdl.

If mdl is not allocated, FltFastIoMdlWriteComplete will raise BSoD.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cbc4c2d0-a210-ce92-4d00-4fa665b18711
* Version Independent ID: dd7c0fa8-4aec-7c5a-e73c-29f94c88be8a
* Content: [FltFastIoPrepareMdlWrite function (fltkernel.h) - Windows drivers](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/fltkernel/nf-fltkernel-fltfastiopreparemdlwrite)
* Content Source: [wdk-ddi-src/content/fltkernel/nf-fltkernel-fltfastiopreparemdlwrite.md](https://github.com/MicrosoftDocs/windows-driver-docs-ddi/blob/staging/wdk-ddi-src/content/fltkernel/nf-fltkernel-fltfastiopreparemdlwrite.md)
* Product: **windows-hardware**
* Technology: **windows-devices**
* GitHub Login: @lorihollasch
* Microsoft Alias: **loriwhip**

**Comment by hsebs** [comment]
my test code

```C
...

status = FltCreateFileEx(
	gFileFltHandle,
	DestinationFileInstance,
	&fileHandle,
	&fileObject,
	GENERIC_WRITE,
	&objectAttribute,
	&statusBlock,
	&fileSize,
	FILE_ATTRIBUTE_NORMAL,
	0,
	FILE_CREATE,
	FILE_NON_DIRECTORY_FILE,
	NULL,
	0,
	0);

if (NT_SUCCESS(status))
{
	offset.QuadPart = 0;
	mdl = (PMDL)1;
	if (FltFastIoPrepareMdlWrite(DestinationFileInstance,
		fileObject,
		&offset,
		CLUSTER_SIZE,
		0,
		&mdl,
		&statusBlock) == FALSE)
	{
		KdPrint(("FltFastIoPrepareMdlWrite : Failed\n"));
	}
	STATUS_FLT_DISALLOW_FAST_IO
	if(FltFastIoMdlWriteComplete(DestinationFileInstance, fileObject, &offset, mdl) == FALSE)
	{
		KdPrint(("FltFastIoMdlWriteComplete : Failed\n"));
	}

	sObDereferenceObject(fileObject);
	FltClose(fileHandle);
}

...
```

If `statusBlock.information` is `STATUS_FLT_DISALLOW_FAST_IO` then `mdl` is not modified.
But `FltFastIoMdlWriteComplete` raises BSoD when an invalid MDL value is entered.

**Comment by hsebs** [comment]
It is fixed in #967
